src/app/core/services/code.service.spec.ts

```diff
@@ -47,11 +47,33 @@ describe('CodeService', () => {
     expect(service.structure.next).toHaveBeenCalledWith(talk);
   });
 
+  it('expects visibilty flagged key to be discarded', async () =>{
```

Comment:
I have fixed points 1 and 3.

I'll be honest, I am not sure I understand point 2 fully.

When you say 
> If you use the spyOn(service.structure, 'next').and.stub(); code, then you can simply check that.

> simplify the expect in your code

Are you suggesting any of the following?

1.  I should add `spyOn(service.structure, 'next').and.stub();` and then only check for `expect(service.structure.next).toHaveBeenCalledWith(talk);` ?
**In this case we won't be able to check the `visibility` functionality at all.**

2. I should add `spyOn(service.structure, 'next').and.stub();` and keep the rest of the code as it is.
**In this case we stub the call, then check for the object contents. The test fails as you'd expect it to.**

3. I should keep the existing code, and also check for the `spyOn(service.structure, 'next').and.stub();` call afterwards. (Not sure if this can be done within the same `it` structure.
**I don't see the purpose of this as the previous unit tests covers it completely.**

Reply:
I would go with choice 1. It does check the visibility functionality - if you pass three slides: one with no `visibility` key/value, one with `visibility: true`, and one with `visibility: false`, the expected array passed to `.next` would contain two slides (the first and second) which tests your added code.

Reply:
I get it now.
I had misunderstood the purpose of the `toHaveBeenCalledWith()` call. 
Will rewrite immediately. Thanks!